nntrainer/layers/layer_node.h

```diff
@@ -309,11 +332,15 @@ class LayerNode : public ml::train::Layer, public GraphNode {
   friend std::ostream &operator<<(std::ostream &out, const LayerNode &l);
 
 private:
-  // TODO: make this unique_ptr once getObject API is removed
+  /// @todo: remove this
   std::shared_ptr<nntrainer::LayerV1>
+    layerv1; /**< The actual object in the graph node */
+
+  std::unique_ptr<nntrainer::Layer>
```

Comment:
how about a union of layerv1 and layer? both should not exist together.

Reply:
I was thinking `std::variant` but even in this case we need to check which one exist so,  we have to add another member, I though this is not a big advantage.

Reply:
We will need some `finalized` kind of variable to track which context to pass to `setBatch()` (check #1312) which sits on top of this PR itself :).

Reply:
Yeah, this will be deleted anyway so I guess it's fine, from the user's perspective they are only able to set only one regulated from the constructor.

Reply:
cool.